libs/vkd3d/command.c

```diff
@@ -6339,6 +6340,49 @@ static void STDMETHODCALLTYPE d3d12_command_list_CopyTextureRegion(d3d12_command
         return;
 
     d3d12_command_list_ensure_batch(list, copy_info.batch_type);
+    alias = false;
+    for (size_t i = 0; i < list->transfer_batch.batch_len; i++)
+    {
+        const struct vkd3d_image_copy_info *other_info = &list->transfer_batch.batch[i];
+        const VkImageSubresourceLayers *subres, *other_subres;
+        switch (copy_info.batch_type)
+        {
+            case VKD3D_BATCH_TYPE_COPY_BUFFER_TO_IMAGE:
+                // Testing for destination aliasing is sufficient to avoid both duplicate layout transitions
+                // and WAW hazards.
+                subres = &copy_info.copy.buffer_image.imageSubresource;
+                other_subres = &other_info->copy.buffer_image.imageSubresource;
+                alias |= copy_info.dst.pResource == other_info->dst.pResource &&
+                         subres->aspectMask == other_subres->aspectMask &&
+                         subres->baseArrayLayer == other_subres->baseArrayLayer &&
+                         subres->mipLevel == other_subres->mipLevel;
+                break;
+            case VKD3D_BATCH_TYPE_COPY_IMAGE_TO_BUFFER:
+                // Test for source aliasing, needed to avoid issuing duplicate layout transitions which would
```

Comment:
Source aliasing is fine. When we add image barriers, we'll avoid duplicate transitions anyways. I only think we need to consider aliasing for BUFFER_TO_IMAGE.